product_customer_code_sale/models/sale.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2017 Agile Business Group sagl
+#     (<http://www.agilebg.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.multi
+    def _compute_get_product_customer_code(self):
+        product_supplierinfo_obj = self.env['product.supplierinfo']
+        for line in self.filtered(lambda sol: sol.product_id.supplier_ids):
+            product = line.product_id
+            code_id = product_supplierinfo_obj.search([
```

Comment:
This search is bad. The customer and supplierinfo of type 'customer' should be included as criteria.